<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>cisMR-cML for robust cis Mendelian randomization analysis</title>

<script>// Pandoc 2.9 adds attributes on both header and div. We remove the former (to
// be compatible with the behavior of Pandoc < 2.8).
document.addEventListener('DOMContentLoaded', function(e) {
  var hs = document.querySelectorAll("div.section[class*='level'] > :first-child");
  var i, h, a;
  for (i = 0; i < hs.length; i++) {
    h = hs[i];
    if (!/^h[1-6]$/i.test(h.tagName)) continue;  // it should be a header h1-h6
    a = h.attributes;
    while (a.length > 0) h.removeAttribute(a[0].name);
  }
});
</script>

<style type="text/css">
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
span.underline{text-decoration: underline;}
div.column{display: inline-block; vertical-align: top; width: 50%;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
</style>



<style type="text/css">
code {
white-space: pre;
}
.sourceCode {
overflow: visible;
}
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
{ counter-reset: source-line 0; }
pre.numberSource code > span
{ position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
{ content: counter(source-line);
position: relative; left: -1em; text-align: right; vertical-align: baseline;
border: none; display: inline-block;
-webkit-touch-callout: none; -webkit-user-select: none;
-khtml-user-select: none; -moz-user-select: none;
-ms-user-select: none; user-select: none;
padding: 0 4px; width: 4em;
color: #aaaaaa;
}
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa; padding-left: 4px; }
div.sourceCode
{ }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } 
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.at { color: #7d9029; } 
code span.bn { color: #40a070; } 
code span.bu { color: #008000; } 
code span.cf { color: #007020; font-weight: bold; } 
code span.ch { color: #4070a0; } 
code span.cn { color: #880000; } 
code span.co { color: #60a0b0; font-style: italic; } 
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.do { color: #ba2121; font-style: italic; } 
code span.dt { color: #902000; } 
code span.dv { color: #40a070; } 
code span.er { color: #ff0000; font-weight: bold; } 
code span.ex { } 
code span.fl { color: #40a070; } 
code span.fu { color: #06287e; } 
code span.im { color: #008000; font-weight: bold; } 
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.kw { color: #007020; font-weight: bold; } 
code span.op { color: #666666; } 
code span.ot { color: #007020; } 
code span.pp { color: #bc7a00; } 
code span.sc { color: #4070a0; } 
code span.ss { color: #bb6688; } 
code span.st { color: #4070a0; } 
code span.va { color: #19177c; } 
code span.vs { color: #4070a0; } 
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } 
</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    var j = 0;
    while (j < rules.length) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") {
        j++;
        continue;
      }
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') {
        j++;
        continue;
      }
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>




<style type="text/css">body {
background-color: #fff;
margin: 1em auto;
max-width: 700px;
overflow: visible;
padding-left: 2em;
padding-right: 2em;
font-family: "Open Sans", "Helvetica Neue", Helvetica, Arial, sans-serif;
font-size: 14px;
line-height: 1.35;
}
#TOC {
clear: both;
margin: 0 0 10px 10px;
padding: 4px;
width: 400px;
border: 1px solid #CCCCCC;
border-radius: 5px;
background-color: #f6f6f6;
font-size: 13px;
line-height: 1.3;
}
#TOC .toctitle {
font-weight: bold;
font-size: 15px;
margin-left: 5px;
}
#TOC ul {
padding-left: 40px;
margin-left: -1.5em;
margin-top: 5px;
margin-bottom: 5px;
}
#TOC ul ul {
margin-left: -2em;
}
#TOC li {
line-height: 16px;
}
table {
margin: 1em auto;
border-width: 1px;
border-color: #DDDDDD;
border-style: outset;
border-collapse: collapse;
}
table th {
border-width: 2px;
padding: 5px;
border-style: inset;
}
table td {
border-width: 1px;
border-style: inset;
line-height: 18px;
padding: 5px 5px;
}
table, table th, table td {
border-left-style: none;
border-right-style: none;
}
table thead, table tr.even {
background-color: #f7f7f7;
}
p {
margin: 0.5em 0;
}
blockquote {
background-color: #f6f6f6;
padding: 0.25em 0.75em;
}
hr {
border-style: solid;
border: none;
border-top: 1px solid #777;
margin: 28px 0;
}
dl {
margin-left: 0;
}
dl dd {
margin-bottom: 13px;
margin-left: 13px;
}
dl dt {
font-weight: bold;
}
ul {
margin-top: 0;
}
ul li {
list-style: circle outside;
}
ul ul {
margin-bottom: 0;
}
pre, code {
background-color: #f7f7f7;
border-radius: 3px;
color: #333;
white-space: pre-wrap; 
}
pre {
border-radius: 3px;
margin: 5px 0px 10px 0px;
padding: 10px;
}
pre:not([class]) {
background-color: #f7f7f7;
}
code {
font-family: Consolas, Monaco, 'Courier New', monospace;
font-size: 85%;
}
p > code, li > code {
padding: 2px 0px;
}
div.figure {
text-align: center;
}
img {
background-color: #FFFFFF;
padding: 2px;
border: 1px solid #DDDDDD;
border-radius: 3px;
border: 1px solid #CCCCCC;
margin: 0 5px;
}
h1 {
margin-top: 0;
font-size: 35px;
line-height: 40px;
}
h2 {
border-bottom: 4px solid #f7f7f7;
padding-top: 10px;
padding-bottom: 2px;
font-size: 145%;
}
h3 {
border-bottom: 2px solid #f7f7f7;
padding-top: 10px;
font-size: 120%;
}
h4 {
border-bottom: 1px solid #f7f7f7;
margin-left: 8px;
font-size: 105%;
}
h5, h6 {
border-bottom: 1px solid #ccc;
font-size: 105%;
}
a {
color: #0033dd;
text-decoration: none;
}
a:hover {
color: #6666ff; }
a:visited {
color: #800080; }
a:visited:hover {
color: #BB00BB; }
a[href^="http:"] {
text-decoration: underline; }
a[href^="https:"] {
text-decoration: underline; }

code > span.kw { color: #555; font-weight: bold; } 
code > span.dt { color: #902000; } 
code > span.dv { color: #40a070; } 
code > span.bn { color: #d14; } 
code > span.fl { color: #d14; } 
code > span.ch { color: #d14; } 
code > span.st { color: #d14; } 
code > span.co { color: #888888; font-style: italic; } 
code > span.ot { color: #007020; } 
code > span.al { color: #ff0000; font-weight: bold; } 
code > span.fu { color: #900; font-weight: bold; } 
code > span.er { color: #a61717; background-color: #e3d2d2; } 
</style>



<style type="text/css">
.main-container {
max-width: 940px;
margin-left: auto;
margin-right: auto;
}
img {
max-width:100%;
}
.tabbed-pane {
padding-top: 12px;
}
.html-widget {
margin-bottom: 20px;
}
button.code-folding-btn:focus {
outline: none;
}
summary {
display: list-item;
}
details > summary > p:only-child {
display: inline;
}
pre code {
padding: 0;
}
</style>



<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
display: inline-table;
max-height: 500px;
min-height: 44px;
overflow-y: auto;
border: 1px solid #ddd;
border-radius: 4px;
}
.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
content: "\e259";
font-family: 'Glyphicons Halflings';
display: inline-block;
padding: 10px;
border-right: 1px solid #ddd;
}
.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
content: "\e258";
font-family: 'Glyphicons Halflings';
border: none;
}
.tabset-dropdown > .nav-tabs > li.active {
display: block;
}
.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
border: none;
display: inline-block;
border-radius: 4px;
background-color: transparent;
}
.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
display: block;
float: none;
}
.tabset-dropdown > .nav-tabs > li {
display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div id="header">



<h1 class="title toc-ignore">cisMR-cML for robust cis Mendelian
randomization analysis</h1>

</div>


<p>This vignette shows an example of applying cisMR-cML to infer causal
relationship of LDL on CAD using cis-SNPs on the PCSK9 region.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a><span class="fu">library</span>(cisMRcML)</span></code></pre></div>
<div id="example" class="section level2">
<h2>Example</h2>
<p>We provide the pre-processed data for LDL and CAD on the PCSK9
region.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;mr_dat&quot;</span>)</span>
<span id="cb2-2"><a href="#cb2-2" tabindex="-1"></a><span class="fu">str</span>(mr_dat)</span>
<span id="cb2-3"><a href="#cb2-3" tabindex="-1"></a><span class="co">#&gt; List of 7</span></span>
<span id="cb2-4"><a href="#cb2-4" tabindex="-1"></a><span class="co">#&gt;  $ exp_df:Classes &#39;data.table&#39; and &#39;data.frame&#39;: 9 obs. of  12 variables:</span></span>
<span id="cb2-5"><a href="#cb2-5" tabindex="-1"></a><span class="co">#&gt;   ..$ SNP   : chr [1:9] &quot;rs55862049&quot; &quot;rs187607506&quot; &quot;rs11206510&quot; &quot;rs17111503&quot; ...</span></span>
<span id="cb2-6"><a href="#cb2-6" tabindex="-1"></a><span class="co">#&gt;   ..$ A1    : chr [1:9] &quot;A&quot; &quot;C&quot; &quot;T&quot; &quot;A&quot; ...</span></span>
<span id="cb2-7"><a href="#cb2-7" tabindex="-1"></a><span class="co">#&gt;   ..$ A2    : chr [1:9] &quot;C&quot; &quot;T&quot; &quot;C&quot; &quot;G&quot; ...</span></span>
<span id="cb2-8"><a href="#cb2-8" tabindex="-1"></a><span class="co">#&gt;   ..$ freq  : num [1:9] 0.853 0.755 0.812 0.732 0.982 ...</span></span>
<span id="cb2-9"><a href="#cb2-9" tabindex="-1"></a><span class="co">#&gt;   ..$ b     : num [1:9] -0.0309 0.0419 0.0418 -0.0373 0.3003 ...</span></span>
<span id="cb2-10"><a href="#cb2-10" tabindex="-1"></a><span class="co">#&gt;   ..$ se    : num [1:9] 0.00289 0.00242 0.00262 0.00232 0.00773 ...</span></span>
<span id="cb2-11"><a href="#cb2-11" tabindex="-1"></a><span class="co">#&gt;   ..$ p     : num [1:9] 1.33e-26 4.30e-67 2.23e-57 3.06e-58 0.00 ...</span></span>
<span id="cb2-12"><a href="#cb2-12" tabindex="-1"></a><span class="co">#&gt;   ..$ N     : int [1:9] 343621 343621 343621 343621 343621 343621 343621 343621 343621</span></span>
<span id="cb2-13"><a href="#cb2-13" tabindex="-1"></a><span class="co">#&gt;   ..$ z     : num [1:9] -10.7 17.3 16 -16.1 38.8 ...</span></span>
<span id="cb2-14"><a href="#cb2-14" tabindex="-1"></a><span class="co">#&gt;   ..$ cor   : num [1:9] -0.0182 0.0295 0.0272 -0.0274 0.0661 ...</span></span>
<span id="cb2-15"><a href="#cb2-15" tabindex="-1"></a><span class="co">#&gt;   ..$ se_cor: num [1:9] 0.00171 0.00171 0.00171 0.00171 0.0017 ...</span></span>
<span id="cb2-16"><a href="#cb2-16" tabindex="-1"></a><span class="co">#&gt;   ..$ bJ    : num [1:9] -0.01173 0.01487 -0.00377 -0.01608 0.06235 ...</span></span>
<span id="cb2-17"><a href="#cb2-17" tabindex="-1"></a><span class="co">#&gt;   ..- attr(*, &quot;.internal.selfref&quot;)=&lt;externalptr&gt; </span></span>
<span id="cb2-18"><a href="#cb2-18" tabindex="-1"></a><span class="co">#&gt;  $ out_df:Classes &#39;data.table&#39; and &#39;data.frame&#39;: 9 obs. of  12 variables:</span></span>
<span id="cb2-19"><a href="#cb2-19" tabindex="-1"></a><span class="co">#&gt;   ..$ SNP   : chr [1:9] &quot;rs55862049&quot; &quot;rs187607506&quot; &quot;rs11206510&quot; &quot;rs17111503&quot; ...</span></span>
<span id="cb2-20"><a href="#cb2-20" tabindex="-1"></a><span class="co">#&gt;   ..$ A1    : chr [1:9] &quot;A&quot; &quot;C&quot; &quot;T&quot; &quot;A&quot; ...</span></span>
<span id="cb2-21"><a href="#cb2-21" tabindex="-1"></a><span class="co">#&gt;   ..$ A2    : chr [1:9] &quot;C&quot; &quot;T&quot; &quot;C&quot; &quot;G&quot; ...</span></span>
<span id="cb2-22"><a href="#cb2-22" tabindex="-1"></a><span class="co">#&gt;   ..$ freq  : num [1:9] 0.868 0.804 0.848 0.736 0.986 ...</span></span>
<span id="cb2-23"><a href="#cb2-23" tabindex="-1"></a><span class="co">#&gt;   ..$ b     : num [1:9] -0.0301 0.0636 0.0745 -0.0349 0.2565 ...</span></span>
<span id="cb2-24"><a href="#cb2-24" tabindex="-1"></a><span class="co">#&gt;   ..$ se    : num [1:9] 0.0143 0.0129 0.0133 0.0113 0.0573 ...</span></span>
<span id="cb2-25"><a href="#cb2-25" tabindex="-1"></a><span class="co">#&gt;   ..$ p     : num [1:9] 3.48e-02 7.67e-07 2.34e-08 2.01e-03 7.47e-06 ...</span></span>
<span id="cb2-26"><a href="#cb2-26" tabindex="-1"></a><span class="co">#&gt;   ..$ N     : int [1:9] 184305 184305 184305 184305 184305 184305 184305 184305 184305</span></span>
<span id="cb2-27"><a href="#cb2-27" tabindex="-1"></a><span class="co">#&gt;   ..$ z     : num [1:9] -2.11 4.94 5.58 -3.09 4.48 ...</span></span>
<span id="cb2-28"><a href="#cb2-28" tabindex="-1"></a><span class="co">#&gt;   ..$ cor   : num [1:9] -0.00492 0.01151 0.01301 -0.00719 0.01043 ...</span></span>
<span id="cb2-29"><a href="#cb2-29" tabindex="-1"></a><span class="co">#&gt;   ..$ se_cor: num [1:9] 0.00233 0.00233 0.00233 0.00233 0.00233 ...</span></span>
<span id="cb2-30"><a href="#cb2-30" tabindex="-1"></a><span class="co">#&gt;   ..$ bJ    : num [1:9] -0.00236 0.00286 0.00777 -0.00347 0.00774 ...</span></span>
<span id="cb2-31"><a href="#cb2-31" tabindex="-1"></a><span class="co">#&gt;   ..- attr(*, &quot;.internal.selfref&quot;)=&lt;externalptr&gt; </span></span>
<span id="cb2-32"><a href="#cb2-32" tabindex="-1"></a><span class="co">#&gt;  $ N1    : int 343621</span></span>
<span id="cb2-33"><a href="#cb2-33" tabindex="-1"></a><span class="co">#&gt;  $ N2    : num 184305</span></span>
<span id="cb2-34"><a href="#cb2-34" tabindex="-1"></a><span class="co">#&gt;  $ LD_mat: num [1:9, 1:9] 1 -0.2278 -0.1979 0.022 -0.0396 ...</span></span>
<span id="cb2-35"><a href="#cb2-35" tabindex="-1"></a><span class="co">#&gt;   ..- attr(*, &quot;dimnames&quot;)=List of 2</span></span>
<span id="cb2-36"><a href="#cb2-36" tabindex="-1"></a><span class="co">#&gt;   .. ..$ : chr [1:9] &quot;rs55862049&quot; &quot;rs187607506&quot; &quot;rs11206510&quot; &quot;rs17111503&quot; ...</span></span>
<span id="cb2-37"><a href="#cb2-37" tabindex="-1"></a><span class="co">#&gt;   .. ..$ : chr [1:9] &quot;rs55862049&quot; &quot;rs187607506&quot; &quot;rs11206510&quot; &quot;rs17111503&quot; ...</span></span>
<span id="cb2-38"><a href="#cb2-38" tabindex="-1"></a><span class="co">#&gt;  $ exp_IV: chr [1:8] &quot;rs55862049&quot; &quot;rs187607506&quot; &quot;rs17111503&quot; &quot;rs11591147&quot; ...</span></span>
<span id="cb2-39"><a href="#cb2-39" tabindex="-1"></a><span class="co">#&gt;  $ out_IV: chr &quot;rs11206510&quot;</span></span></code></pre></div>
<p>As shown above, 9 SNPs in the PCSK9 region were used in the analysis
(the first real data application in the manuscript), in which 8 SNPs
(shown in <code>mr_dat$exp_IV</code>) were jointly associated (selected
by <a href="https://yanglab.westlake.edu.cn/software/gcta/#COJO">GCTA-COJO</a>)
with LDL, and 1 SNP (shown in <code>mr_dat$out_IV</code>) was associate
with CAD. <code>mr_dat$LD_mat</code> is the LD/correlation matrix among
the 9 SNPs, which was estimated based on the UK Biobank White-ancestry
individuals.</p>
<p><code>mr_dat$exp_df</code> and <code>mr_dat$out_df</code> correspond
to the GWAS summary statistics for LDL (exposure) and CAD (outcome)
respectively. In particular, <code>mr_dat$exp_df$b</code>,
<code>mr_dat$exp_df$se</code>, <code>mr_dat$exp_df$p</code> are the raw
GWAS marginal summary statistics, and <code>mr_dat$exp_df$bJ</code> is
the <strong>conditional</strong> genetic effect estimates on the
exposure (similar for <code>mr_dat$out_df</code>), which can be
calculated as follows (see Manuscript Section 4.2):</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" tabindex="-1"></a>mr_dat<span class="sc">$</span>exp_df<span class="sc">$</span>cor <span class="ot">=</span> mr_dat<span class="sc">$</span>exp_df<span class="sc">$</span>b <span class="sc">/</span> <span class="fu">sqrt</span>(mr_dat<span class="sc">$</span>exp_df<span class="sc">$</span>b<span class="sc">^</span><span class="dv">2</span> <span class="sc">+</span> (mr_dat<span class="sc">$</span>exp_df<span class="sc">$</span>N <span class="sc">-</span> <span class="dv">2</span>) <span class="sc">*</span></span>
<span id="cb3-2"><a href="#cb3-2" tabindex="-1"></a>                                             mr_dat<span class="sc">$</span>exp_df<span class="sc">$</span>se<span class="sc">^</span><span class="dv">2</span>)</span>
<span id="cb3-3"><a href="#cb3-3" tabindex="-1"></a>mr_dat<span class="sc">$</span>exp_df<span class="sc">$</span>bJ <span class="ot">=</span> <span class="fu">solve</span>(mr_dat<span class="sc">$</span>LD_mat) <span class="sc">%*%</span> mr_dat<span class="sc">$</span>exp_df<span class="sc">$</span>cor</span>
<span id="cb3-4"><a href="#cb3-4" tabindex="-1"></a>mr_dat<span class="sc">$</span>out_df<span class="sc">$</span>cor <span class="ot">=</span> mr_dat<span class="sc">$</span>out_df<span class="sc">$</span>b <span class="sc">/</span> <span class="fu">sqrt</span>(mr_dat<span class="sc">$</span>out_df<span class="sc">$</span>b<span class="sc">^</span><span class="dv">2</span> <span class="sc">+</span> (mr_dat<span class="sc">$</span>out_df<span class="sc">$</span>N <span class="sc">-</span> <span class="dv">2</span>) <span class="sc">*</span></span>
<span id="cb3-5"><a href="#cb3-5" tabindex="-1"></a>                                             mr_dat<span class="sc">$</span>out_df<span class="sc">$</span>se<span class="sc">^</span><span class="dv">2</span>)</span>
<span id="cb3-6"><a href="#cb3-6" tabindex="-1"></a>mr_dat<span class="sc">$</span>out_df<span class="sc">$</span>bJ <span class="ot">=</span> <span class="fu">solve</span>(mr_dat<span class="sc">$</span>LD_mat) <span class="sc">%*%</span> mr_dat<span class="sc">$</span>out_df<span class="sc">$</span>cor</span></code></pre></div>
<p>Next, we apply cisMR-cML with 100 data perturbations with 5 random
starts where <span class="math inline">\(\theta^{(0)}\sim
U(-0.1,0.1)\)</span>:</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" tabindex="-1"></a>b_exp_cond<span class="ot">=</span>mr_dat<span class="sc">$</span>exp_df<span class="sc">$</span>bJ</span>
<span id="cb4-2"><a href="#cb4-2" tabindex="-1"></a>b_out_cond<span class="ot">=</span>mr_dat<span class="sc">$</span>out_df<span class="sc">$</span>bJ </span>
<span id="cb4-3"><a href="#cb4-3" tabindex="-1"></a>Sig_exp1 <span class="ot">=</span> <span class="fu">solve</span>(mr_dat<span class="sc">$</span>LD_mat) <span class="sc">%*%</span> (mr_dat<span class="sc">$</span>exp_df<span class="sc">$</span>se_cor <span class="sc">%o%</span> mr_dat<span class="sc">$</span>exp_df<span class="sc">$</span>se_cor <span class="sc">*</span> mr_dat<span class="sc">$</span>LD_mat) <span class="sc">%*%</span></span>
<span id="cb4-4"><a href="#cb4-4" tabindex="-1"></a>  <span class="fu">solve</span>(mr_dat<span class="sc">$</span>LD_mat)</span>
<span id="cb4-5"><a href="#cb4-5" tabindex="-1"></a>Sig_out1 <span class="ot">=</span> <span class="fu">solve</span>(mr_dat<span class="sc">$</span>LD_mat) <span class="sc">%*%</span> (mr_dat<span class="sc">$</span>out_df<span class="sc">$</span>se_cor <span class="sc">%o%</span> mr_dat<span class="sc">$</span>out_df<span class="sc">$</span>se_cor <span class="sc">*</span> mr_dat<span class="sc">$</span>LD_mat) <span class="sc">%*%</span></span>
<span id="cb4-6"><a href="#cb4-6" tabindex="-1"></a>  <span class="fu">solve</span>(mr_dat<span class="sc">$</span>LD_mat)</span>
<span id="cb4-7"><a href="#cb4-7" tabindex="-1"></a>Sig_exp_inv<span class="ot">=</span><span class="fu">solve</span>(Sig_exp1)</span>
<span id="cb4-8"><a href="#cb4-8" tabindex="-1"></a>Sig_out_inv<span class="ot">=</span><span class="fu">solve</span>(Sig_out1)</span>
<span id="cb4-9"><a href="#cb4-9" tabindex="-1"></a></span>
<span id="cb4-10"><a href="#cb4-10" tabindex="-1"></a>ciscML_res <span class="ot">=</span> <span class="fu">cismr_cML_DP</span>(<span class="at">b_exp=</span>b_exp_cond,<span class="at">b_out=</span>b_out_cond,</span>
<span id="cb4-11"><a href="#cb4-11" tabindex="-1"></a>                          <span class="at">Sig_exp_inv=</span>Sig_exp_inv,<span class="at">Sig_out_inv=</span>Sig_out_inv,<span class="at">maxit=</span><span class="dv">200</span>,</span>
<span id="cb4-12"><a href="#cb4-12" tabindex="-1"></a>                          <span class="at">n =</span> mr_dat<span class="sc">$</span>N1,<span class="at">random_start =</span> <span class="dv">5</span>,</span>
<span id="cb4-13"><a href="#cb4-13" tabindex="-1"></a>                          <span class="at">min_theta_range=</span><span class="sc">-</span><span class="fl">0.1</span>,<span class="at">max_theta_range=</span><span class="fl">0.1</span>,</span>
<span id="cb4-14"><a href="#cb4-14" tabindex="-1"></a>                          <span class="at">num_pert=</span><span class="dv">100</span>,<span class="at">random_start_pert=</span><span class="dv">5</span>,<span class="at">random_seed =</span> <span class="dv">12345</span>)</span></code></pre></div>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;The estimated effect of the exposure on outcome: &quot;</span>, ciscML_res<span class="sc">$</span>BIC_DP_theta)</span>
<span id="cb5-2"><a href="#cb5-2" tabindex="-1"></a><span class="co">#&gt; The estimated effect of the exposure on outcome:  0.1845757</span></span>
<span id="cb5-3"><a href="#cb5-3" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;Standard error of the causal estimate: &quot;</span>, ciscML_res<span class="sc">$</span>BIC_DP_se)</span>
<span id="cb5-4"><a href="#cb5-4" tabindex="-1"></a><span class="co">#&gt; Standard error of the causal estimate:  0.05461205</span></span>
<span id="cb5-5"><a href="#cb5-5" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;P-value for the causal estimate: &quot;</span>, ciscML_res<span class="sc">$</span>BIC_DP_p)</span>
<span id="cb5-6"><a href="#cb5-6" tabindex="-1"></a><span class="co">#&gt; P-value for the causal estimate:  0.0007254868</span></span></code></pre></div>
<p>At this point, we have reproduced the LDL-CAD result in our
manuscript.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" tabindex="-1"></a><span class="fu">sessionInfo</span>()</span>
<span id="cb6-2"><a href="#cb6-2" tabindex="-1"></a><span class="co">#&gt; R version 3.6.3 (2020-02-29)</span></span>
<span id="cb6-3"><a href="#cb6-3" tabindex="-1"></a><span class="co">#&gt; Platform: x86_64-apple-darwin15.6.0 (64-bit)</span></span>
<span id="cb6-4"><a href="#cb6-4" tabindex="-1"></a><span class="co">#&gt; Running under: OS X  14.3</span></span>
<span id="cb6-5"><a href="#cb6-5" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb6-6"><a href="#cb6-6" tabindex="-1"></a><span class="co">#&gt; Matrix products: default</span></span>
<span id="cb6-7"><a href="#cb6-7" tabindex="-1"></a><span class="co">#&gt; LAPACK: /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRlapack.dylib</span></span>
<span id="cb6-8"><a href="#cb6-8" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb6-9"><a href="#cb6-9" tabindex="-1"></a><span class="co">#&gt; locale:</span></span>
<span id="cb6-10"><a href="#cb6-10" tabindex="-1"></a><span class="co">#&gt; [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8</span></span>
<span id="cb6-11"><a href="#cb6-11" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb6-12"><a href="#cb6-12" tabindex="-1"></a><span class="co">#&gt; attached base packages:</span></span>
<span id="cb6-13"><a href="#cb6-13" tabindex="-1"></a><span class="co">#&gt; [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span id="cb6-14"><a href="#cb6-14" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb6-15"><a href="#cb6-15" tabindex="-1"></a><span class="co">#&gt; other attached packages:</span></span>
<span id="cb6-16"><a href="#cb6-16" tabindex="-1"></a><span class="co">#&gt; [1] cisMRcML_0.0.0.9000</span></span>
<span id="cb6-17"><a href="#cb6-17" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb6-18"><a href="#cb6-18" tabindex="-1"></a><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span id="cb6-19"><a href="#cb6-19" tabindex="-1"></a><span class="co">#&gt;  [1] digest_0.6.35       MASS_7.3-51.4       R6_2.5.1            lifecycle_1.0.4    </span></span>
<span id="cb6-20"><a href="#cb6-20" tabindex="-1"></a><span class="co">#&gt;  [5] jsonlite_1.8.8      evaluate_0.23       cachem_1.0.8        rlang_1.1.3        </span></span>
<span id="cb6-21"><a href="#cb6-21" tabindex="-1"></a><span class="co">#&gt;  [9] cli_3.6.2           rstudioapi_0.15.0   jquerylib_0.1.4     bslib_0.6.1        </span></span>
<span id="cb6-22"><a href="#cb6-22" tabindex="-1"></a><span class="co">#&gt; [13] rmarkdown_2.26      tools_3.6.3         numDeriv_2016.8-1.1 xfun_0.42          </span></span>
<span id="cb6-23"><a href="#cb6-23" tabindex="-1"></a><span class="co">#&gt; [17] yaml_2.3.8          fastmap_1.1.1       compiler_3.6.3      htmltools_0.5.7    </span></span>
<span id="cb6-24"><a href="#cb6-24" tabindex="-1"></a><span class="co">#&gt; [21] knitr_1.45          sass_0.4.8</span></span></code></pre></div>
</div>




</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
